pyquil/api/_qcs_client.py

```diff
@@ -28,7 +29,18 @@ def qcs_client(*, client_configuration: QCSClientConfiguration, request_timeout:
     :param client_configuration: Client configuration.
     :param request_timeout: Time limit for requests, in seconds.
     """
+    _ensure_event_loop()
     with build_sync_client(
         configuration=client_configuration, client_kwargs={"timeout": request_timeout}
     ) as client:  # type: httpx.Client
         yield client
+
+
+def _ensure_event_loop() -> None:
+    try:
+        asyncio.get_event_loop()
+    except RuntimeError as ex:
+        if len(ex.args) > 0 and "There is no current event loop in thread" in ex.args[0]:
```

Comment:
😬

Reply:
Yeah, unfortunately they don't use an error subtype :(

Reply:
I thought this what [get_event_loop](https://docs.python.org/3/library/asyncio-eventloop.html?highlight=current%20event%20loop%20thread#asyncio.get_event_loop) does as opposed to [get_running_loop](https://docs.python.org/3/library/asyncio-eventloop.html?highlight=current%20event%20loop%20thread#asyncio.get_running_loop)?

If not the case, is it worth defining a custom [event loop policy](https://docs.python.org/3/library/asyncio-policy.html#asyncio-policies)?

Reply:
It looks like `get_running_loop` has to be called from a coroutine only, so not sure we can use it.

On the custom policy: If the behavior we needed were more complex, this would probably be the best approach. I'll punt for now, as what we have works and is fairly simple, but it's something to keep an eye on if we hit issues with the current implementation.